### PR TITLE
Remove `BOOTMAGIC_ENABLE = lite` option

### PR DESCRIPTION
**Pull request by noroadsleft**
## Description

Final stage of the `BOOTMAGIC_ENABLE` rework.

Changes all instances of `BOOTMAGIC_ENABLE = lite` to `BOOTMAGIC_ENABLE = yes`.

## Types of Changes

- [x] Core
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
  - The Japanese docs will need an update here. I looked them over and it seems the English docs have diverged as far as Bootmagic functionality is concerned, but some of the updates required look to be out of scope for this PR.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by tzarc — APPROVED** [review]
Burn it with fire.

**Comment by noroadsleft** [comment]
@mtei,

I would appreciate guidance on the changes I made to the Japanese docs in [`a6de112`](https://github.com/qmk/qmk_firmware/pull/15002/commits/a6de11215a6a2c2da1c1e314067c4cede52f2ea0).

----

@qmk/python,

Similarly, I'm not sure if the changes in [`8d16db1`](https://github.com/qmk/qmk_firmware/pull/15002/commits/8d16db16fba18b00c560345b8beab9fe2896701d) and [`ed4afda`](https://github.com/qmk/qmk_firmware/pull/15002/commits/ed4afda76671a383303912838bb728144b897213) encompass everything that's needed for the data-driven side of this equation.

File: docs/ja/config_options.md [hunk 1]
```diff
@@ -358,7 +358,7 @@ QMK での全ての利用可能な設定にはデフォルトがあります。
 これらを使って特定の機能のビルドを有効または無効にします。有効にすればするほどファームウェアが大きくなり、MCU には大きすぎるファームウェアを構築するリスクがあります。
 
 * `BOOTMAGIC_ENABLE`
-  * 仮想 DIP スイッチ設定
+  * Bootmagic Lite を有効にする
```
Comment (mtei) [inline docs/ja/config_options.md:361]: For consistency with the rest of the translation, do the following.
```suggestion
  * ブートマジックライトを有効にします
```

File: docs/ja/faq_misc.md [hunk 2]
```diff
@@ -39,10 +39,6 @@ Size after:
 
 **NKRO** がまだ動作しない場合は、`Magic` **N** コマンド(デフォルトでは `LShift+RShift+N`)を試してみてください。**NKRO** モードと **6KRO** モード間を一時的に切り替えるためにこのコマンドを使うことができます。**NKRO** が機能しない状況、特に BIOS の場合は **6KRO** モードに切り替える必要があります。
 
-ファームウェアを `BOOTMAGIC_ENABLE` でビルドした場合、`ブートマジック` **N** コマンドで切り替える必要があります(デフォルトでは `Space+N`)。この設定は EEPROM に格納され、電源を入れ直しても保持されます。
```
Comment (mtei) [inline docs/ja/faq_misc.md:42]: This paragraph still seems to be in docs/faq_misc.md. Doesn't that also need to be updated?

File: docs/ja/config_options.md (hunk 1 above)
Comment (noroadsleft) [reply docs/ja/config_options.md]: Edited in [`ec4dbb9`](https://github.com/qmk/qmk_firmware/pull/15002/commits/ec4dbb9725242bec4b70559de8f248ca07444296).

File: docs/ja/faq_misc.md (hunk 2 above)
Comment (noroadsleft) [reply docs/ja/faq_misc.md:42]: Updated in [`f4ae08c`](https://github.com/qmk/qmk_firmware/pull/15002/commits/f4ae08cffec5ccc07fa9ac5b1959f071ec9febf7)

**Review by drashna — APPROVED** [review]

**Review by mtei — APPROVED** [review]

**Pull request merged**